pkg/reconciler/common/workload_override.go

```diff
@@ -183,6 +184,11 @@ func replaceProbes(override *base.WorkloadOverride, ps *corev1.PodTemplateSpec)
 					FailureThreshold:              override.FailureThreshold,
 					TerminationGracePeriodSeconds: override.TerminationGracePeriodSeconds,
 				}
+				if *overrideProbe == (v1.Probe{}) {
```

Comment:
I am not sure if my logic got something wrong:
Line 179 overrideProbe := &v1.Probe{...}, *overrideProbe is for sure equal to (v1.Probe{}). so Line 192 will never reach.

Reply:
L192 will still reachable.

- The new logic happens only when users didn't specify any override probe. (=`overrideProbe` is empty.)
- L192 will be reached when users specify some override probe && the target manifest does not have any probe. (=`overrideProbe` is not empty but the manifest's `containers[i].ReadinessProbe` is empty.)